test/symmetries/fusiontrees.jl

```diff
@@ -33,54 +33,63 @@ using .TestSetup
-            @constinferred FusionTree((), u, ())
-            @constinferred FusionTree((u,), u, (false,))
-            @constinferred FusionTree((u, u), u, (false, false))
-            @constinferred FusionTree((u, u, u), u)
-            @constinferred FusionTree((u, u, u, u))
-            @test_throws MethodError FusionTree((u, u), u, (false, false, false))
-        else
-            errstr = "fusion tree requires inner lines if `FusionStyle(I) <: MultipleFusion`"
-            @test_throws errstr FusionTree((), u, ())
-            @test_throws errstr FusionTree((u,), u, (false,))
-            @test_throws errstr FusionTree((u, u), u, (false, false))
-            @test_throws errstr FusionTree((u, u, u), u)
-            @test_throws errstr FusionTree((u, u, u, u))
+        for u in allunits(I)
+            @constinferred FusionTree((), u, (), (), ())
+            @constinferred FusionTree((u,), u, (false,), (), ())
+            @constinferred FusionTree((u, u), u, (false, false), (), (1,))
+            @constinferred FusionTree((u, u, u), u, (false, false, false), (u,), (1, 1))
+            @constinferred FusionTree(
+                (u, u, u, u), u, (false, false, false, false), (u, u), (1, 1, 1)
+            )
+            @test_throws MethodError FusionTree((u, u, u), u, (false, false), (u,), (1, 1))
+            @test_throws MethodError FusionTree(
+                (u, u, u), u, (false, false, false), (u, u), (1, 1)
+            )
+            @test_throws MethodError FusionTree(
+                (u, u, u), u, (false, false, false), (u,), (1, 1, 1)
+            )
+            @test_throws MethodError FusionTree((u, u, u), u, (false, false, false), (), (1,))
+
+            f = FusionTree((u, u, u), u, (false, false, false), (u,), (1, 1))
+            @test sectortype(f) == I
+            @test length(f) == 3
+            @test FusionStyle(f) == FusionStyle(I)
+            @test BraidingStyle(f) == BraidingStyle(I)
+
+            if FusionStyle(I) isa UniqueFusion
+                @constinferred FusionTree((), u, ())
+                @constinferred FusionTree((u,), u, (false,))
+                @constinferred FusionTree((u, u), u, (false, false))
+                @constinferred FusionTree((u, u, u), u)
+                if isa(UnitStyle(I), SimpleUnit)
+                    @constinferred FusionTree((u, u, u, u))
+                end
+                @test_throws MethodError FusionTree((u, u), u, (false, false, false))
+            else
+                errstr = "fusion tree requires inner lines if `FusionStyle(I) <: MultipleFusion`"
+                @test_throws errstr FusionTree((), u, ())
+                @test_throws errstr FusionTree((u,), u, (false,))
+                @test_throws errstr FusionTree((u, u), u, (false, false))
+                @test_throws errstr FusionTree((u, u, u), u)
+                if isa(UnitStyle(I), SimpleUnit)
+                    @test_throws errstr FusionTree((u, u, u, u))
+                end
```

Comment:
Can you actually replace the `error("...")` on line 95 of src/fusiontrees/fusiontrees.jl with a `throw(ArgumentError("..."))` and replace these tests with `@test_throws ArgumentError FusionTree(...)`. I think that is nicer than testing the specific error message, since we might want to tune these error messages a bit better, and so these tests are very brittle. 

Maybe then the last statement `@test_throws ArgumentError FusionTree((u, u, u, u))` is true for both `SimpleUnit` (because of the `MultipleFusion` and the lack of inner lines) and `GenericUnit` (because the lack of a single unit for the coupled sector).